### ISSUE TITLE
Add support for MemoryDB as Idemopotency backend

### ISSUE THREAD

**Issue by whardier** [awslabs/aws-lambda-powertools-python#629]
https://aws.amazon.com/about-aws/whats-new/2021/08/amazon-memorydb-redis/

**Comment by gmcrocetti** [comment]
I'm putting myself available to tackle this issue.

**Comment by heitorlessa** [comment]
That would be awesome! I think vanilla Redis makes more sense than MemoryDB
as it’s still early to know how many customers would want it.

FYI - I’m back on Sep 27th and will review the PR by then.

Muito obrigado Guilherme ;-)

**Comment by danielloader** [comment]
Isn't it meant to be the same client implementation for both?

**Comment by heitorlessa** [comment]
Yep, though I haven’t had a chance to look into it yet for any caveats.

**Comment by heitorlessa** [comment]
Just read the docs, you’re right Daniel, it’s fully compatible — we could
use redis-py-cluster as an optional dependency. Depending on the
implementation, it’d be easy to also provide Redis support for Parameters
and Feature Flag.

https://github.com/Grokzen/redis-py-cluster

**Comment by whardier** [comment]
Redis in general would be fairly rad to have available via an extra requirement.  I wasn't entirely sure if a broad implementation in aws-lambda-powertools for redis support would ultimately support both ElasticCache and MemoryDB.

**Comment by gmcrocetti** [comment]
@heitorlessa . Things have changed, won't have bandwidth to tackle this issue :/. I hope someone else will volunteer. Enjoy your time ! Obrigado :+1:
